src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -665,6 +755,27 @@ private List<SyntaxNode> VerifyAndPermuteParamNodes(IEnumerable<XmlElementSyntax
         }
 
         protected override IEnumerable<AbstractFormattingRule> GetFormattingRules(Document document)
-            => SpecializedCollections.SingletonEnumerable(new ChangeSignatureFormattingRule()).Concat(Formatter.GetDefaultFormattingRules(document));
+            => Formatter.GetDefaultFormattingRules(document).Concat(new ChangeSignatureFormattingRule());
+
+        protected override SyntaxNode AddNameToArgument(SyntaxNode newArgument, string name)
+        {
+            return (newArgument as ArgumentSyntax).WithNameColon(NameColon(name));
+        }
+
+        protected override SyntaxNode CreateExplicitParamsArrayFromIndividualArguments(SeparatedSyntaxList<SyntaxNode> newArguments, int indexInExistingList, IParameterSymbol parameterSymbol)
+        {
+            var listOfArguments = SeparatedList(newArguments.Skip(indexInExistingList).Select(a => ((ArgumentSyntax)a).Expression), newArguments.GetSeparators().Skip(indexInExistingList));
+            var initializerExpression = InitializerExpression(SyntaxKind.ArrayInitializerExpression, listOfArguments);
+            var objectCreation = ArrayCreationExpression((ArrayTypeSyntax)parameterSymbol.Type.GenerateTypeSyntax(), initializerExpression);
```

Comment:
i don't know if this ArrayTypeSyntax cast is safe in error scenarios.  it may crash.

Reply:
IsParamsArrayExpanded is false in error scenarios, so it should never get in here. I've added an assert that the parameter symbol we have here is IsParams.

Reply:
👍